### ISSUE TITLE
Build error on macOS: `IOV_MAX` undefined

### ISSUE THREAD

**Issue by eschnett** [open-mpi/ompi#12110]
I am building OpenMPI 5.0.0 on macOS with a MacPorts-installed GCC 13.2.

My configure options are

$ ./configure --prefix=$HOME/openmpi-5.0.0 --with-libevent=internal CC=gcc CXX=g++ FC=gfortran
This fails with the error message
```
fbtl_posix_preadv.c: In function 'mca_fbtl_posix_preadv_generic':
fbtl_posix_pwritev.c: In function 'mca_fbtl_posix_pwritev_generic':
fbtl_posix_preadv.c:282:30: error: 'IOV_MAX' undeclared (first use in this function); did you mean 'INT_MAX'?
  282 |                 (iov_count < IOV_MAX ) ){
      |                              ^~~~~~~
      |                              INT_MAX
fbtl_posix_pwritev.c:286:31: error: 'IOV_MAX' undeclared (first use in this function); did you mean 'INT_MAX'?
  286 |                  (iov_count < IOV_MAX )) {
      |                               ^~~~~~~
      |                               INT_MAX
fbtl_posix_preadv.c:282:30: note: each undeclared identifier is reported only once for each function it appears in
fbtl_posix_pwritev.c:286:31: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by jsquyres** [comment]
Oddly, with Homebrew-installed GCC/G++/Gfortran 13.2.0, I do not have the compile problems reported here.

```
$ uname -a
Darwin Little-Beezle.local 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020 arm64

$ gcc-13 --version
gcc-13 (Homebrew GCC 13.2.0) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++-13 --version
g++-13 (Homebrew GCC 13.2.0) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ gfortran-13 --version
GNU Fortran (Homebrew GCC 13.2.0) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by bosilca** [comment]
I use macport to build gcc13 (gcc 13.2.0_4 compiled with only the basic variant (stdlib)). This is the only version that I have and has these issues. I tried gcc10, gcc12, clang15 (both macport and apple), llvm (15, 16 and devel).

I found multiple software in github that have protections against the lack of IOV_MAX.

**Comment by wenduwan** [comment]
Re #12108 #12109 @bosilca Do you plan to backport the fixes to 5.0.x?

**Comment by jsquyres** [comment]
> Re #12108 #12109 @bosilca Do you plan to backport the fixes to 5.0.x?

PR #12267 filed to back port the fixes to v5.0.x.

**Comment by wenduwan** [comment]
Backported. Closing. Thanks @bosilca @jsquyres
